packages/materialized/src/controls/material-input.control.tsx

```diff
@@ -1,32 +1,49 @@
 import * as React from 'react';
 import {
   computeLabel,
+  Control,
+  ControlElement,
   ControlProps,
+  ControlState,
   DispatchField,
   isControl,
+  isDescriptionHidden,
   mapStateToControlProps,
   rankWith,
-  registerStartupRenderer
+  registerStartupRenderer,
+  resolveSchema
 } from 'jsonforms-core';
 import { connect } from 'react-redux';
 
 import { InputLabel } from 'material-ui/Input';
 import { FormControl, FormHelperText } from 'material-ui/Form';
 
-export const MaterialInputControl =
-  ({ classNames, id, errors, label, uischema, schema, visible, required }: ControlProps) => {
-  const isValid = errors.length === 0;
-  const trim = uischema.options && uischema.options.trim;
+export class MaterialInputControl extends Control<ControlProps, ControlState> {
+  render() {
+    const { classNames, id, errors, label, uischema, schema, visible, required } = this.props;
+    const isValid = errors.length === 0;
+    const trim = uischema.options && uischema.options.trim;
+    const controlElement = uischema as ControlElement;
+    const resolvedSchema = resolveSchema(schema, controlElement.scope.$ref);
+    const description = resolvedSchema.description === undefined ? '' : resolvedSchema.description;
 
-  return (
-    <FormControl className={classNames.wrapper} hidden={!visible} fullWidth={!trim}>
-      <InputLabel htmlFor={id} className={classNames.label}>
-        {computeLabel(label, required)}
-      </InputLabel>
-      <DispatchField uischema = {uischema} schema = {schema}/>
-      <FormHelperText error={!isValid}>{errors}</FormHelperText>
-    </FormControl>
-  );
+    return (
+      <FormControl
+        className={classNames.wrapper}
+        hidden={!visible}
+        fullWidth={!trim}
+        onFocus={() => this.onFocus()}
+        onBlur={() => this.onBlur()}
+      >
+        <InputLabel htmlFor={id} className={classNames.label}>
+          {computeLabel(label, required)}
+        </InputLabel>
+        <DispatchField uischema={uischema} schema={schema}/>
+        <FormHelperText error={!isValid}>{errors}</FormHelperText>
+        {isDescriptionHidden(visible, description, this.state.isFocused) ? null : <FormHelperText error={false}>{description}</FormHelperText>}
```

Comment:
can't we mere the two FormHelperText into one or is it more common to have two FormHelperText elements?

Reply:
I read helper and error text patterns on this article `https://material.io/guidelines/patterns/errors.html#errors-incompatible-state-errors`, so they just use one FormHelperText. I will make the changes on the existing FormHelperText instead of creating a new one